### PR TITLE
Adding format input field for date/time.

### PR DESCRIPTION
**Pull request by danbev**
Motivation:
When adding a Datestamp the format of the data/time should be
configuration which is it currently not. This commit should add such a
field.

Modifications:
Added a field labed 'Format' and gave it a default value of YYYY-MM-DD
HH:MM.

Result:
When displayed in fh-ngui there is now a an input field for specifying
the format that the date/time should be in.

JIRA:
https://issues.jboss.org/browse/RHMAP-4771

### CODE REVIEW

**Comment by danbev** [comment]
@nialldonnellyfh When you get a chance could you take a look at this and see if this is what you had in mind for [RHMAP-4771](https://issues.jboss.org/browse/RHMAP-4771)?

File: coffee/fields/autodate.coffee [hunk 1]
```diff
@@ -23,9 +23,9 @@ Formbuilder.registerField 'autodate',
             <option value="datetime">Date &amp; Time</option>
             <option value="time">Time Only</option>
             <option value="date">Date Only</option>
-          </select>
-          <!--<label>Auto-populate:</label>
-          <input type='checkbox' data-rv-checked='model.<%= Formbuilder.options.mappings.TIME_AUTOPOPULATE  %>' />-->
+          </select><br />
+          <label>Format:</label>
+          <input type="text" value="YYYY-MM-DD HH:MM" style="width: auto;">
```
Comment (nialldonnellyfh) [inline coffee/fields/autodate.coffee]: The format input needs to be mapped to the field model.

See https://github.com/fheng/formbuilder/pull/14/files#diff-6662bf93ae7fee50eb47affe17eac973L22

Comment (nialldonnellyfh) [inline coffee/fields/autodate.coffee]: I'm using fieldOptions.definition.dateTimeFormat as the format option. Can you use this when associating with the model.

File: coffee/fields/autodate.coffee [hunk 2]
```diff
@@ -23,9 +23,9 @@ Formbuilder.registerField 'autodate',
             <option value="datetime">Date &amp; Time</option>
             <option value="time">Time Only</option>
             <option value="date">Date Only</option>
-          </select>
-          <!--<label>Auto-populate:</label>
-          <input type='checkbox' data-rv-checked='model.<%= Formbuilder.options.mappings.TIME_AUTOPOPULATE  %>' />-->
+          </select><br />
+          <label>Format:</label>
+          <input type="text" data-value="model.<%= Formbuilder.options.mappings.DATETIME_FORMAT %>" style="width: auto;"/>
```
Comment (nialldonnellyfh) [inline coffee/fields/autodate.coffee]: data-rv-input for text field input 

See https://github.com/fheng/formbuilder/blob/master/coffee/fields/text.coffee#L21

**Comment by nialldonnellyfh** [comment]
@danbev Looks good.

**Comment by danbev** [comment]
@nialldonnellyfh Would you mind taking a look at this last commit and see if you think this is ok?

**Comment by nialldonnellyfh** [comment]
@danbev Looks good. feel free to merge to master

**Comment by nialldonnellyfh** [comment]
Closed for https://github.com/fheng/formbuilder/pull/15

**Pull request closed without merging**